keyboards/handwired/orbweaver/keymaps/default/keymap.c

```diff
@@ -0,0 +1,125 @@
+/* Copyright 2023 a_marmot
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see &lt;http://www.gnu.org/licenses/&gt;.
+*/
+
+#include QMK_KEYBOARD_H
+#include <is31fl3731.h>
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    /*
+     * ┌───┌───┬───┬───┬───┐
+     * │Esc│ 1 │ 2 │ 3 │ 4 │
+     * ├───┼───┼───┼───┼───┤
+     * │Tab│ q │ w │ e │ r │
+     * ├───┼───┼───┼───┼───┤
+     * │Ctl│ a │ s │ d │ f │
+     * ├───┼───┼───┼───┼───┤
+     * │Shi│ z │ x │ c │ v │
+     * ├───┼───┼───┼───┼───┤
+     * │Alt│ ↑ │ ↓ │ → │ ← │
+     * ├───┼───┼───┴───┴───┤
+     * │Spa│Key│           │
+     * └───┴───┴───────────┘
+     */
+//base layer is a selection menu and bootload access
+    [0] = LAYOUT_keypad(
+        TO(0), TO(0), TO(0), TO(0), QK_BOOT,
+        TO(0), TO(0), TO(0), TO(0), TO(0),
+        TO(0), TO(0), TO(0), TO(0), TO(0),
+        TO(0), TO(1), TO(2), TO(3), TO(0),
+        KC_LOPT, TO(0), TO(0), TO(0), TO(0),
+        KC_LSFT, TO(0)
```

Comment:
I'd really advise against setting it up like this, as most of these keys aren't gonna do anything. Especially having `QK_BOOT` on the base layer.. sounds like a recipe for confusion for anyone else using this.

Reply:
The default RGB layer has all inactive keys dark.  So when you turn on the power you see illuminated keys of different colors indicating selectable profiles (all others are dark).  The QK_BOOT key is off by itself, in bright red, so impossible to miss.  I prefer it on the base layer since when I am adjusting key maps and colors I need to frequently reflash it.

Reply:
Another arrangement I tried was rotating profiles.  In this case QK_BOOT was on one of the profiles by itself.  I don't like this as much, since it required more use of the mode key on the side of the unit to access desired key maps.  With a menu approach, the base layer contains all possible branches, and the mode key is not required unless you need to return to the main menu.  I think this actually makes it easier and safer for others trying to reproduce this mod, since they only have to mess with profiles 1, 2, 3, etc.  and leave the base layer (menu) alone. It is a pain to reflash if QK_BOOT is accidentally lost.

Reply:
I can send you a video if you like.  It's not like a regular numpad where all the keys are static reflective images.  With the RGB on it's clear as daylight which keys do something, and which are off.

Reply:
Probably this part of the [PR Checklist](https://docs.qmk.fm/#/pr_checklist) applies here:
> - default (and via) keymaps should be “pristine”
>    - bare minimum to be used as a “clean slate” for another user to develop their own user-specific keymap
>    - standard layouts preferred in these keymaps, if possible
>    - default keymap should not enable VIA – the VIA integration documentation requires a keymap called `via`
> - submitters can have a personal (or bells-and-whistles) keymap showcasing capabilities in the same PR but it shouldn’t be embedded in the ‘default’ keymap
> - submitters can also have a “manufacturer-matching” keymap that mirrors existing functionality of the commercial product, if porting an existi"bootmagic": trueng board
So you should provide a simple `default` keymap, and may add your own keymap with whatever custom layouts and lighting code you want.

As for reflashing, you have `"bootmagic": true`, and the `[0, 0]` key in the matrix is the top left one as usual, so the Bootmagic feature should work irrespective of the keymap contents.

Reply:
What if I (as the manufacturer), decide that a principle feature of the device is that the default keymap is a menu to other user-definable keymaps?

Reply:
I just tested bootmagic, and it does indeed seem to insure the MCU is re-flashable even withour QK_BOOT.  So let's do this.  I will just upload a generic version of keymap.c with a single default keymap, and let users do whatever they want.  Sound OK?